src/hacbs/components/ApplicationDetails/tabs/ComponentsTab.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import { Link } from 'react-router-dom';
+import { useK8sWatchResource } from '@openshift/dynamic-plugin-sdk-utils';
+import {
+  EmptyState,
+  EmptyStateIcon,
+  Title,
+  EmptyStateBody,
+  Button,
+  Bullseye,
+  Spinner,
+  EmptyStateVariant,
+} from '@patternfly/react-core';
+import { OutlinedFileImageIcon } from '@patternfly/react-icons/dist/esm/icons/outlined-file-image-icon';
+import ComponentListView from '../../../../components/ApplicationDetailsView/ComponentListView';
+import { ComponentGroupVersionKind } from '../../../../models';
+import { ComponentKind } from '../../../../types';
+
+type ComponentTabProps = {
+  applicationName: string;
+  namespace: string;
+};
+
+const ComponentsTab: React.FC<ComponentTabProps> = ({ applicationName, namespace }) => {
+  const [components, componentsLoaded] = useK8sWatchResource<ComponentKind[]>({
+    groupVersionKind: ComponentGroupVersionKind,
+    namespace,
+    isList: true,
+  });
+
+  const filteredComponents = React.useMemo(
+    () =>
+      componentsLoaded ? components?.filter((c) => c.spec.application === applicationName) : [],
+    [components, applicationName, componentsLoaded],
+  );
+
+  const componentList = (
+    <ComponentListView applicationName={applicationName} components={filteredComponents} />
+  );
+  const loading = (
+    <Bullseye>
+      <Spinner />
+    </Bullseye>
+  );
+
```

Comment:
nit: the spinner appears at the top of the content area
![image](https://user-images.githubusercontent.com/14068621/177573444-5c96b6f0-e977-4fa9-a2c9-5639ee7f9a4b.png)

Reply:
Fixed it.